classes/autoupdate.php

```diff
@@ -44,7 +44,20 @@ public static function update_from_event(\core\event\base $event) : void {
                         $i->completion == COMPLETION_TRACKING_AUTOMATIC &&
                         $i->instance != $data['contextinstanceid']
                     ) {
-                        $completion->update_state($i, COMPLETION_UNKNOWN, $data['userid']);
+                        if ($i->groupmode > 0) {
+                            $group = groups_get_activity_group($i);
+                        }
+                        if (isset($group) && $group) {
+                            $members = groups_get_members($group);
+                        }
+                        if (!isset($members) || !$members) {
```

Comment:
using `empty` here would be a bit easier to understand.

Reply:
maybe also an inline comment that this is also covering the case of no group mode would be nice